phoenix-core/src/main/java/org/apache/phoenix/mapreduce/PhoenixInputFormat.java

```diff
@@ -185,7 +184,7 @@ private QueryPlan getQueryPlan(final JobContext context, final Configuration con
                 scan.setAttribute(BaseScannerRegionObserver.TX_SCN, Bytes.toBytes(Long.valueOf(txnScnValue)));
             }
             // Initialize the query plan so it sets up the parallel scans
-            queryPlan.iterator(MapReduceParallelScanGrouper.getInstance());
+            queryPlan.iterator(TableMRParallelScanGrouper.init(configuration));
```

Comment:
Can't this just instantiate a TableMRParallelScanGrouper and the init method can be invoked in the constructor so that it doesn't need to be exposed in the interface?